index.ts

```diff
@@ -1,28 +1,74 @@
-import express from "express";
-import mongoose from "mongoose";
-import { json, urlencoded } from "express";
+const mongoose = require('mongoose')
+const cors = require('cors');
+const dotenv = require('dotenv')
+const express = require('express');
+const { auth, requiresAuth } = require('express-openid-connect');
+
 import routes from "./routes";
-import cors from "cors";
 
-const dotenv = require("dotenv");
-dotenv.config();
+const app = express();
+
+// load the .env file if the server isn’t in production mode
+if (process.env.NODE_ENV !== 'production') {
+  dotenv.config();
+}
+
+const config = {
+  authRequired: false,
+  auth0Logout: true,
+  secret: 'a long, randomly-generated string stored in env',
+  baseURL: 'http://localhost:8080',
+  clientID: 'UAdvr0ooyHVrALfpjLSHJhzelNDPXvjR',
+  issuerBaseURL: 'https://dev-w205xag6.us.auth0.com'
+};
+
+// auth router attaches /login, /logout, and /callback routes to the baseURL
+app.use(auth(config));
+
+// req.isAuthenticated is provided from the auth router
+app.get('/', (req, res) => {

```

Comment:
may want to put / and /profile in routes folder so they are all in one place, unless of course these are required to be in the top level file